File: app/src/main/java/com/example/blizzard/model/OpenWeatherApi.java
package com.example.blizzard.model;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

/**
 * Created by kelvi on 8/3/2020
 */
public interface OpenWeatherApi {
    @GET("weather?")
    Call<WeatherData> getWeatherByCityName(
            @Query("q") String cityName,
            @Query("appid") String apiKey
    );

    @GET("weather?")
    Call<WeatherData> getWeatherByLongitudeLatitude(
            @Query("lat") Double latitude,
            @Query("lon") Double longitude,
            @Query("appid") String apiKey
    );

}

File: app/src/main/java/com/example/blizzard/data/api/OpenWeatherApi.kt
package com.example.blizzard.data.api

import com.example.blizzard.model.WeatherDataResponse
import retrofit2.http.GET
import retrofit2.http.Query

/**
 * Created by kelvi on 8/3/2020
 */
interface OpenWeatherApi {
    @GET("weather?")
    suspend fun getWeatherByCityName(
            @Query("q") cityName: String?,
            @Query("appid") apiKey: String?
    ): WeatherDataResponse

    @GET("weather?")
    suspend fun getWeatherByLongitudeLatitude(
            @Query("lat") latitude: Double?,
            @Query("lon") longitude: Double?,
            @Query("appid") apiKey: String?
    ): WeatherDataResponse
}